src/snowflake/snowpark/session.py

```diff
@@ -1853,11 +1853,10 @@ def table(self, name: Union[str, Iterable[str]]) -> Table:
             >>> session.table([current_db, current_schema, "my_table"]).collect()
             [Row(A=1, B=2), Row(A=3, B=4)]
         """
-
         if not isinstance(name, str) and isinstance(name, Iterable):
             name = ".".join(name)
         validate_object_name(name)
-        t = Table(name, self)
```

Comment:
will any expectation tests be added within this PR?

Reply:
Good question: I actually had a question about `Session.table(...)` vs `Table(...)` here:

`Session.table()` is already used by all other expectation tests, so no new coverage is necessary, at least not until we start depending in some way on the fact that the `SpTable` instance came from `Session.table()` (as opposed to another API call).

Does anything ever create `Table` objects directly? The `__init__` function has at least one peculiarity, as it can take a `None` `session` argument. Is it ever idiomatic in Pandas, or other libraries that have dataframes, to instantiate `Table` (or `Table`-like) objects directly?

Reply:
It seems to be documented: https://docs.snowflake.com/developer-guide/snowpark/reference/python/latest/snowpark/api/snowflake.snowpark.Table

Reply:
The documentation says:

> You can create a [Table](https://docs.snowflake.com/developer-guide/snowpark/reference/python/latest/snowpark/api/snowflake.snowpark.Table#snowflake.snowpark.Table) object by calling [Session.table()](https://docs.snowflake.com/developer-guide/snowpark/reference/python/latest/snowpark/api/snowflake.snowpark.Session.table.html#snowflake.snowpark.Session.table) with the name of the table in Snowflake. See examples in [Session.table()](https://docs.snowflake.com/developer-guide/snowpark/reference/python/latest/snowpark/api/snowflake.snowpark.Session.table.html#snowflake.snowpark.Session.table).

Does it ever make sense to instantiate it standalone?

Reply:
Just to be clear, I think we have everything we need to round trip `df = Table("foo")`. I'm just unsure whether this is worth testing.

Reply:
I think it is.

Reply:
Done

Comes with matching unparser change: https://github.com/snowflakedb/snowflake/pull/192287